### PR TITLE
Add support for setting package mirrors

### PR DESCRIPTION
**Pull request by wallyworld**
The package commander interface gets a SetMirrorCommands() method which returns the commands needed to update the mirrors for the package manager. This is currently implemented for apt.

As a driveby, the `gopkg.in/retry.v1` dependency is replaced with `juju/retry`. This significantly reduces the use of older, indirect dependencies. The main impact is to change the `RunCommandWithRetry` method to conform to the slightly different implementation of the new retry package.

### CODE REVIEW

File: config/apt_constants.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ const (
 	// currently configured APT source location. We assume that
 	// the first source for "main" in the file is the one that
 	// should be replaced throughout the file.
-	ExtractAptSource = `awk "/^deb .* $(lsb_release -sc) .*main.*\$/{print \$2;exit}" ` + AptSourcesFile
```
Comment (wallyworld) [inline config/apt_constants.go:27]: `lsb_release` command is not always available, eg minimal Ubuntu image used for k8s jujud agent

**Review by ycliuhw — APPROVED** [review]
LGTM, thanks

**Comment by wallyworld** [comment]
$$merge$$

**Comment by erik78se** [comment]
How about centos repos with yum?

**Pull request merged**